geniza/corpus/admin.py

```diff
@@ -94,6 +95,17 @@ def get_queryset(self, request):
         return super().get_queryset(request) \
             .prefetch_related('tags', 'languages', 'textblock_set')
 
+    def save_model(self, request, obj, form, change):
+        '''Customize this model's save_model function and then execute the
+         existing admin.ModelAdmin save_model function'''
+        if '_saveasnew' in request.POST:
+            # Get the ID from the admin URL
+            original_pk = resolve(request.path).kwargs['object_id']
+            # Get the original object
+            original_doc = obj._meta.concrete_model.objects.get(id=original_pk)
+            clone_message = f'Cloned from {str(original_doc)}'
+            obj.notes = obj.notes + '\n' + clone_message if obj.notes else clone_message
```

Comment:
Please add a test case where document has existing notes

Reply:
Testing `notes` revealed that the object was not properly cloned in the first place. I tried debugging, but fixing this isn't straightforward. I can put an hour or so into this next week if you'd like me to ask Kate.